src/gen_rules.ml

```diff
@@ -1225,6 +1225,32 @@ module Gen(P : Params) = struct
          ~dir
          ~targets)
 
+  let alias_rules (alias_conf : Alias_conf.t) ~dir =
+    let digest =
+      let source =
+        match alias_conf.action with
+        | None -> ""
+        | Some a -> Sexp.to_string (User_action.Unexpanded.sexp_of_t a) in
+      Digest.to_hex (Digest.string source) in
+    let digest_path = Path.of_string (alias_conf.name ^ digest) in
```

Comment:
This should be `Path.relative ~dir (alias_conf.name ^ digest)`.

For the basename we should use the same file prefix as `Alias.make`, so we could have: `Path.basename (Alias.file alias) ^ "-" ^ digest`